pkg/skaffold/tag/input_digest_test.go

```diff
@@ -28,16 +28,18 @@ import (
 func TestInputDigest_GenerateCorrectChecksumForSingleFile(t *testing.T) {
 	testutil.Run(t, "", func(t *testutil.T) {
 		dir := t.TempDir()
-		d1 := []byte("hello\ngo\n")
-		filePath := filepath.Join(dir, "temp.file")
-		ioutil.WriteFile(filePath, d1, 0644)
-
-		hash, _ := fileHasher(filePath)
-
-		// because we are hashing content of file and it's path
-		// we can't get a stable hash in testing because call t.TempDir()
-		// will return a folder to a random name
-		re := regexp.MustCompile(`^[a-fA-F0-9]{32}$`)
-		t.CheckTrue(re.MatchString(hash))
+		cwdBackup, err := os.Getwd()
+		t.RequireNoError(err)
+		t.RequireNoError(os.Chdir(dir))
+		defer func() { t.RequireNoError(os.Chdir(cwdBackup)) }()
```

Comment:
If we create d1 and d2 with the same contents, then the fileHash of both should be equal.  Currently this would fail since we're hashing the full paths of the files in d1 and d2.

Typically Skaffold doesn't change the working directory — it rather executes programs with a different working directory.

Reply:
The scenario that broke us before was that when calculating the tag for an image defined in the "root" skaffold config all paths were relative to the workspace directory. Even the workspace directory path was relative, i.e. just ".". When calculating the same image tag when imported via the `requires` section all paths, including the workspace path, were absolute. For the calculation to work at all in the first case _cwd_ must be the directory of the skaffold config or we can't read the file contents.

So this is the scenario I'm trying to test here, that providing paths relative to the workspace directory to the digest calculation will yield the same result as when providing absolute paths. And I can't come up with any other way of getting the relative calculation to work without changing _cwd_.

I can add another test case that verifies that calculating for two different absolute paths works as expected to if you want.

Reply:
I added more tests, including one that tests two different absolute paths and would have failed without the fix.

I would rather leave the test we are discussing here as it is a different case that is closer to the situation were this hit us initially, but if you feel that it should be removed I can do so.